server/src/test/java/org/elasticsearch/persistent/TestPersistentTasksPlugin.java

```diff
@@ -309,10 +309,15 @@ protected void nodeOperation(AllocatedPersistentTask task, TestParams params, Ta
                 AtomicInteger phase = new AtomicInteger();
                 while (true) {
                     // wait for something to happen
-                    assertTrue(awaitBusy(() -> testTask.isCancelled() ||
-                                    testTask.getOperation() != null ||
-                                    clusterService.lifecycleState() != Lifecycle.State.STARTED,   // speedup finishing on closed nodes
-                            30, TimeUnit.SECONDS)); // This can take a while during large cluster restart
+                    try {
+                        assertBusy(() -> assertTrue(testTask.isCancelled() ||
+                                testTask.getOperation() != null ||
+                                clusterService.lifecycleState() != Lifecycle.State.STARTED),   // speedup finishing on closed nodes
+                                30, TimeUnit.SECONDS); // This can take a while during large cluster restart
+                    } catch (Exception e) {
+                        task.markAsFailed(e);
```

Comment:
could you explain why this is needed?

Reply:
This is unnecessary. I have reverted it.